src/User/Controller/SettingsController.php

```diff
@@ -539,4 +558,39 @@ protected function disconnectSocialNetwork($id)
         $account->delete();
         $this->trigger(SocialNetworkConnectEvent::EVENT_AFTER_DISCONNECT, $event);
     }
+
+    public function actionTwoFactorMobilePhone($id)
+    {
+        Yii::$app->response->format = Response::FORMAT_JSON;
+
+        /**
+        * 
+        *
+        * @var User $user 
+        */
+        $user = $this->userQuery->whereId($id)->one();
+
+        if (null === $user) {
+            return [
+                'success' => false,
+                'message' => Yii::t('usuario', 'User not found.')
+            ];
+        }
+        $mobilePhone = Yii::$app->request->get('mobilephone');
+        $currentMobilePhone = $user->getAuthTfMobilePhone();
+        $success=false;
+        if($currentMobilePhone==$mobilePhone){
+            $success=true;    
```

Comment:
This behavior needs to be reviewed, if you have disabled the 2fa and try to re-enabled it with the same phone number, the text message is not sent